apps/vaporgui/VaporWidgets.h

```diff
@@ -0,0 +1,112 @@
+#ifndef VAPORWIDGETS_H 
+#define VAPORWIDGETS_H 
+
+#include <QWidget>
+#include <QDoubleValidator>
+
+#include "Combo.h"
+#include "ui_VPushButton.h"
+#include "ui_VComboBox.h"
+#include "ui_VCheckBox.h"
+#include "ui_VPathSelector.h"
+
+namespace Ui {
+class QSliderEdit;
+}
+
+
+class VPushButton: public QWidget, public Ui_VPushButton
+{
+    Q_OBJECT
+
+public:
+    VPushButton(
+        QWidget* parent, 
+        std::string labelText = "Label",
+        std::string buttonText = "Button"
+    );
+
+    void SetLabelText(  const std::string text );
+    void SetLabelText(  const QString text );
+    void SetButtonText( const std::string text );
+    void SetButtonText( const QString text );
+
+signals:
+    void _pressed();
+
+private slots:
+    void _buttonPressed();
+};
+
+
+
+class VComboBox : public QWidget, public Ui_VComboBox
+{
+    Q_OBJECT
+
+public:
+    VComboBox(
+        QWidget* parent,
+        std::string labelText = "Label"
+    );
+    void        SetLabelText(  const std::string text );
+    int         GetCurrentIndex() const;
+    std::string GetCurrentText() const;
+    void        AddOption( std::string option, int index=0 );
+    void        RemoveOption( int index );
+
+private slots:
+    void _userIndexChanged(int index);
+
+signals:
+    void _indexChanged(int index);
+};
+
+
+
+class VCheckBox : public QWidget, public Ui_VCheckBox
+{
+    Q_OBJECT
+
+public:
+    VCheckBox(
+        QWidget* parent,
+        std::string labelText = "Label"
+    );
+    void SetLabelText( std::string text );
+    bool GetCheckState() const;
+
+private slots:
+    void _userClickedCheckbox();
+
+signals:
+    void _checkboxClicked();
+};
+
+class VPathSelector : public QWidget, public Ui_VPathSelector
+{
+    Q_OBJECT
+
+public:
+    VPathSelector(
+        QWidget* parent,
+        std::string labelText = "Label",
+        std::string filePath = QDir::homePath().toStdString()
+    );
+    void SetLabelText( std::string text );
+    std::string GetPath() const;
+    void SetPath( std::string defaultPath);
+
+private slots:
+    void _openFileDialog();
+    void _setFilePath();
+
+signals:
+    void _pathChanged();
+
+private:
+    std::string _filePath;
+};
+
+
+#endif // VAPORWIDGETS_H
```

Comment:
A few questions regarding a `PathSelector`:
1. Does/Should it support selecting both a folder and a file?
2. Does/Should it support selecting both a single file/folder and multiple files/folders?
3. Does/Should it support file type filters? 
4. Does/Should it check file permissions upon a selection is made?

Reply:
1) I've added file and directory selection options in a recent push
2) I don't think so.  This widget is too small for it.  I can implement another widget that has a QTextEdit so we can display a list of files, but a QLineEdit should really only hold one thing, not a list.
3) I've added this as well, thanks for catching it.
4) Yes, it does for reading files.  Not for writing though.  I'm not certain if we currently need it yet, but if you have a use-case I'll get it in there.

Reply:
I can think of a use case of reading a file: the flow renderer will need to be able to read in a list of seeding points. Can we add this capability now?

Reply:
Yup.